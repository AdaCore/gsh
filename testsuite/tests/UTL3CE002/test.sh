echo "duplicate and rename a file: cp a copy_a"
cp a copy_a
echo "* ls copy_a"
ls copy_a
echo "* cat copy_a"
cat copy_a

rm -r copy_a

echo ""
echo "duplicate and rename a file - use '-f': cp -f a copy_a"
cp -f a copy_a
echo "* ls copy_a"
ls copy_a
echo "* cat copy_a"
cat copy_a

rm -r copy_a

echo ""
echo "duplicate and rename a file - use '-p': cp -p a copy_a"
cp -p a copy_a
echo "* ls copy_a"
ls copy_a
echo "* cat copy_a"
cat copy_a

rm -r copy_a

echo ""
echo "copy a file into an existing directory: cp a dir_a"
cp a dir_a
echo "* ls dir_a/"
ls dir_a/
echo "* dir dir_a/"
cat dir_a/a

rm -r dir_a/a

echo ""
echo "copy and rename a file into an existing directory: cp a dir_a/copy_a"
cp a dir_a/copy_a
echo "* ls dir_a/"
ls dir_a/
echo "* dir dir_a/copy_a"
cat dir_a/copy_a

rm -r dir_a/copy_a

echo ""
echo "copy and rename a file from a dir: cp dir_a/initial copy_a"
cp dir_a/initial copy_a
echo "* ls copy_a"
ls copy_a
echo "* cat copy_a"
cat copy_a

rm -r copy_a

echo ""
echo "duplicate and rename a dir with -r: cp -r dir_a copy_dir_a"
cp -r dir_a copy_dir_a
echo "* ls copy_dir_a/*"
ls copy_dir_a/*
echo "* cat copy_dir_a/*"
cat copy_dir_a/*

rm -r copy_dir_a

echo ""
echo "duplicate and rename a dir with -R: cp -R dir_b copy_dir_b"
cp -R dir_b copy_dir_b
echo "* ls copy_dir_b"
ls copy_dir_b
echo "* cat copy_dir_b/*"
cat copy_dir_b/*

rm -r copy_dir_b

echo ""
echo "copy several files into an existing dir: cp a b.rst dir_a"
cp a b.rst dir_a
echo "* ls dir_a"
ls dir_a
echo "* cat dir_a/*"
cat dir_a/*

rm dir_a/a dir_a/b.rst

echo ""
echo "copy a file and a dir into an existing dir: cp -r a dir_b dir_a"
cp -r a dir_b dir_a
echo "* ls dir_a"
ls dir_a
echo "* cat dir_a/*"
cat dir_a/*

rm -r dir_a/a dir_a/dir_b

echo ""
echo "last copy prevail: cp -r dir_a/* dir_b/* dir_c"
cp  -r dir_a/* dir_b/* dir_c
echo "* ls dir_c"
ls dir_c
echo "* cat dir_c"
cat dir_c/*

rm -r dir_c/initial* dir_c/nested_dir

