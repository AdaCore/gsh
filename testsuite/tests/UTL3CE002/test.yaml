title: Test for cp utility - nominal cases
