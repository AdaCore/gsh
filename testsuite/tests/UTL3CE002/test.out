duplicate and rename a file: cp a copy_a
* ls copy_a
copy_a
* cat copy_a
content of a

duplicate and rename a file - use '-f': cp -f a copy_a
* ls copy_a
copy_a
* cat copy_a
content of a

duplicate and rename a file - use '-p': cp -p a copy_a
* ls copy_a
copy_a
* cat copy_a
content of a

copy a file into an existing directory: cp a dir_a
* ls dir_a/
a
initial
* dir dir_a/
content of a

copy and rename a file into an existing directory: cp a dir_a/copy_a
* ls dir_a/
copy_a
initial
* dir dir_a/copy_a
content of a

copy and rename a file from a dir: cp dir_a/initial copy_a
* ls copy_a
copy_a
* cat copy_a
initial file of dir_a

duplicate and rename a dir with -r: cp -r dir_a copy_dir_a
* ls copy_dir_a/*
copy_dir_a/initial
* cat copy_dir_a/*
initial file of dir_a

duplicate and rename a dir with -R: cp -R dir_b copy_dir_b
* ls copy_dir_b
initial
initial_2
initial_3
nested_dir
* cat copy_dir_b/*
initial_content of dir b
initial_content 2 of dir b


initial_content 3 of dir b
cat: copy_dir_b/nested_dir: Is a directory

copy several files into an existing dir: cp a b dir_a
* ls dir_a
a
b
initial
* cat dir_a/*
content of a
content of b
initial file of dir_a

copy a file and a dir into an existing dir: cp -r a dir_b dir_a
* ls dir_a
a
dir_b
initial
* cat dir_a/*
content of a
cat: dir_a/dir_b: Is a directory
initial file of dir_a

last copy prevail: cp -r dir_a/* dir_b/* dir_c
* ls dir_c
content.txt
initial
initial_2
initial_3
nested_dir
* cat dir_c
initial_content of dir b
initial_content 2 of dir b


initial_content 3 of dir b
cat: dir_c/nested_dir: Is a directory
